platform/persistence/platform-persistence-core-impl/src/main/java/org/codice/ddf/persistence/internal/PersistentStoreImpl.java

```diff
@@ -289,26 +264,28 @@ public int delete(String type, String cql) throws PersistenceException {
         return idsToDelete.size();
     }
 
-    private SolrClient getSolrCore(String storeName) {
-        if (coreSolrClients.containsKey(storeName)) {
-            LOGGER.debug("Returning core {} from map of coreSolrClients", storeName);
-            return coreSolrClients.get(storeName);
+    private SolrClient getSolrClient(String storeName) {
+
+        Future<SolrClient> clientFuture = null;
+        SolrClient solrClient = null;
+
+        if (solrFutures.containsKey(storeName)) {
+            LOGGER.debug("Returning client {} from map of solrFutures", storeName);
+            clientFuture = solrFutures.get(storeName);
+        } else {
+            clientFuture = clientFactory.newClient(storeName);
+            solrFutures.put(storeName, clientFuture);
         }
 
-        // Must specify shard in URL so proper core is used
-        SolrClient coreSolrClient = null;
         try {
-            Future<SolrClient> coreSolrClientFuture =
-                    HttpSolrClientFactory.getHttpSolrClient(solrUrl.getResolvedString(), storeName);
-            coreSolrClient = coreSolrClientFuture.get(30, TimeUnit.SECONDS);
-            coreSolrClients.put(storeName, coreSolrClient);
+            solrClient = clientFuture.get(5, TimeUnit.SECONDS);
 
-            LOGGER.trace("EXITING: getSolrCore");
+            LOGGER.trace("EXITING: getSolrClient");
```

Comment:
Since we don't have a log when entering this method and we don't log anything before L288, should we remove this trace?